docs/cheat-sheets/java-xxe.mdx

```diff
@@ -0,0 +1,341 @@
+    ```
+- `xml-stylesheet` processing instruction: Used to include a stylesheet in an XML document, for example:
+    ```xml
+    <?xml-stylesheet href="include.xsl" type="text/xsl"?>
+    ```
+- XSLT document() function: Used to access nodes in an external XML document, for example:
+    ```xml
+    <xsl:variable name="dummy" select="document('DocumentFunc2.xml')"/>
+    ```
+
+#### References
+
+- [Oracle Docs](https://docs.oracle.com/en/java/javase/13/security/java-api-xml-processing-jaxp-security-guide.html)
+
+### 2.B Security measures
+
+There are a number of security measures, not all of them are available for each class. To see an overview of which measures are available for which class, consult the docs, or go to Section 3 to view the tables with our research results.
+
+#### Feature for Secure Processing (FSP)
+
+Feature for Secure Processing (FSP) is considered the central mechanism for secure XML processing. It is defined as `javax.xml.XMLConstants.FEATURE_SECURE_PROCESSING`.
+For a number of classes this feature is able to protect the parser from all available payloads, but this is not the case for all of the classes we tested.
+
+#### Disabling DTD processing
+
+To disable DTD processing there are a few mechanisms available:
+
+- The `setFeature` method with the following url: `http://apache.org/xml/features/disallow-doctype-decl`. 
+- The `setProperty` method to set `XMLInputFactory.SUPPORT_DTD` to false.
+- The `setAttribute` method to set `XMLConstants.ACCESS_EXTERNAL_DTD` to an empty string (`“”`). 
+
+For some parsers, multiple of these methods are available. 
+
+For most parsers this setting adds additional security for processing XML documents, but it never has any effect on processing schemas or stylesheets. 
+However, there are classes such as `DocumentBuilderFactory`, `Validator`, and `SchemaFactory` where multiple of these methods are available, and setting the above mentioned features throws no exceptions, but only one of them has any effect on the security of the parser.
+
+#### Disabling external schema and stylesheet processing
+
+The `setAttribute` method can be used to set `XMLConstants.ACCESS_EXTERNAL_SCHEMA` and `XMLConstants.ACCESS_EXTERNAL_STYLESHEET` to an empty string to disable these features.
+The effects on security, however, are often limited, with no observed effect for a number of classes.
+
+#### Disabling external general entities and external parameter entities
+
+The  `setFeature` method can be used to set `http://xml.org/sax/features/external-general-entities` and `http://xml.org/sax/features/external-parameter-entities` to false. For the classes where these features are available, the combination of these two protects the parser against all of the tested payloads!
+Except for the `Validator` class, where they have no effect at all.
+There is also `setExpandEntityReferences` that can be set to false and seems to have the same effect as disabling external general entities, based on our testing.
+
```

Comment:
With `http://xml.org/sax/features/external-general-entities` and `http://xml.org/sax/features/external-parameter-entities` - are these methods, calls to some specific URLs - what are they? Can we state this after we list them?